information/specification.md

```diff
@@ -0,0 +1,826 @@
+
+    If a coin selection algorithm is applied to the _complete_ UTxO set of a
+    wallet, then the remaining UTxO set represents the _updated_ UTxO set of the
+    wallet.
+
+## Properties
+
+### Coverage of Payments
+
+This property states that the total value of the [coin
+selection](#coin-selection) result is sufficient to _cover_ the total value of
+the [requested output list](#requested-output-list).
+
+In particular:
+
+  * V<sub>_selected_</sub> ≥ V<sub>_requested_</sub>
+
+Where:
+
+  * V<sub>_requested_</sub>
+
+    is the total value of the [requested output list](#requested-output-list)
+
+
+  * V<sub>_selected_</sub>
+
+    is the total value of the _inputs_ field of the [coin
+    selection](#coin-selection) result.
+
+### Correctness of Change
+
+This property states that the correct amount of _change_ was generated.
+
+In particular:
+
+  * V<sub>_selected_</sub> = V<sub>_requested_</sub> + V<sub>_change_</sub>
+
+Where:
+
+  * V<sub>_change_</sub>
+
+    is the total value of the _change_ field of the [coin
+    selection](#coin-selection) result.
+
+  * V<sub>_requested_</sub>
+
+    is the total value of the [requested output list](#requested-output-list)
+
+  * V<sub>_selected_</sub>
+
+    is the total value of the _inputs_ field of the [coin
+    selection](#coin-selection) result.
+
+### Conservation of UTxO
+
+This property states that every entry in the [initial UTxO
+set](#initial-utxo-set) is included in _either_ the [coin
+selection](#coin-selection) inputs, _or_ in the [remaining UTxO
+set](#remaining-utxo-set), but _not both_.
+
+  * If a UTxO entry _is_ selected by the coin selection algorithm, it is
+    included in the [coin selection](#coin-selection) inputs.
+
+  * If a UTxO entry is _not_ selected by the coin selection algorithm, it is
+    included in the [remaining UTxO set](#remaining-utxo-set).
+
+The following laws hold:
+
+  * UTxO<sub>_selected_</sub> ⊆ UTxO<sub>_initial_</sub>
+  * UTxO<sub>_remaining_</sub> ⊂ UTxO<sub>_initial_</sub>
+  * UTxO<sub>_remaining_</sub> ∩ UTxO<sub>_selected_</sub> = ∅
+  * UTxO<sub>_remaining_</sub> ⋃ UTxO<sub>_selected_</sub> =
+    UTxO<sub>_initial_</sub>
+
+Where:
+
+  * **UTxO<sub>_initial_</sub>**
+
+    is the [initial UTxO set](#initial-utxo-set).
+
+  * **UTxO<sub>_remaining_</sub>**
+
+    is the [remaining UTxO set](#remaining-utxo-set).
+
+  * **UTxO<sub>_selected_</sub>**
+
+    is the value of the _inputs_ field of the [coin selection](#coin-selection)
+    result.
+
+### Conservation of Outputs
+
+This property states that the [requested output list](#requested-output-list)
+is _conserved_ in the [coin selection](#coin-selection) result.
+
+In particular, the _outputs_ field of the [coin selection](#coin-selection)
+result should be _equal to_ the [requested output list](#requested-output-list).
+
+## Failure Modes
+
+There are a number of ways in which a coin selection algorithm can fail:
+
+  * #### UTxO Balance Insufficient
```

Comment:
Does "Not enough money to cover fee" fall into that category, or maybe it could be mentioned as separate one? (Just wondering... :thinking: )

Reply:
Hi @piotr-iohk 
> Does "Not enough money to cover fee" fall into that category, or maybe it could be mentioned as separate one? (Just wondering... thinking )

That's a good point!

I've not yet added the section on fee balancing. But that will include the error you mentioned (as it falls into a separate category of errors).